lib/chef/provisioning/aws_driver/super_lwrp.rb

```diff
@@ -7,30 +7,43 @@ class SuperLWRP < Chef::Resource::LWRPBase
     #
     # Add the :lazy_default and :coerce validation_opts to `attribute`
     #
-    def self.attribute(attr_name, validation_opts={})
-      lazy_default = validation_opts.delete(:lazy_default)
-      coerce = validation_opts.delete(:coerce)
-      if lazy_default || coerce
-        define_method(attr_name) do |arg=nil|
-          arg = instance_exec(arg, &coerce) if coerce && !arg.nil?
+    if self.respond_to?(:properties)
+      # in Chef 12.4+, properties replace attributes and these respond to
+      # coerce and default with a lazy block
+      def self.attribute(attr_name, validation_opts={})
+        lazy_default = validation_opts.delete(:lazy_default)
+        if lazy_default
+          Chef.log_deprecation("#{name} defines a lazy_default validation, instead specify default with a lazy block when Chef supports properties")
```

Comment:
I think I'm trying to do the opposite of that - I don't want to convert `default: lazy { ... }` to `lazy_default`, I want to do the opposite.  I thought thats what I was doing with my current code.

I want to get rid of `lazy_default` in chef-provisioning-aws 2.0 so that we just use the common chef pattern.  What am I missing?